src/components/Modals/Send/hooks/useSendDetails/useSendDetails.tsx

```diff
@@ -109,8 +109,7 @@ export const useSendDetails = (): UseSendDetailsReturnType => {
     state: { wallet },
   } = useWallet()
 
-  const { assetReference } = fromAssetId(assetId)
-  const contractAddress = tokenOrUndefined(assetReference)
+  const contractAddress = contractAddressOrUndefined(assetId)
```

Comment:
suggestion: could rename to `maybeContractAddress` while at it, or make this more declarative and do `isToken` checks on l.131